addons/web/static/src/js/fields/basic_fields_owl.js

```diff
@@ -0,0 +1,131 @@
+odoo.define('web.basic_fields_owl', function (require) {
+    "use strict";
+
+    const AbstractField = require('web.AbstractFieldOwl');
+    const CustomCheckbox = require('web.CustomCheckbox');
+    const core = require('web.core');
+
+    const _lt = core._lt;
+
+    const { useRef } = owl.hooks;
+    const { xml } = owl.tags;
+
+    class FieldBoolean extends AbstractField {
+        /**
+         * @constructor
+         */
+        constructor() {
+            super(...arguments);
+            this.checkboxRef = useRef('checkbox');
+        }
+
+        /**
+         * @param {Object} [nextProps]
+         * @param {Object} [nextProps.record]
+         * @param {Object} [nextProps.event]
+         */
+        willUpdateProps(nextProps) {
+            const prom = super.willUpdateProps(nextProps);
+            if (nextProps.event && nextProps.event.target.name === this.name) {
+                this.activate();
+            }
+            return prom;
+        }
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         * @returns {HTMLElement|null} the focusable checkbox input
+         */
+        get focusableElement() {
+            return this.mode === 'readonly' ? null : this.checkboxRef.comp.input;
+        }
+        /**
+         * A boolean field is always set since false is a valid value.
+         *
+         * @override
+         */
+        get isSet() {
+            return true;
+        }
+        /**
+         * Toggle the checkbox if it is activated due to a click on itself.
+         *
+         * @override
+         */
+        activate(options) {
```

Comment:
not sure to understand the purpose of this function

Reply:
it's for the editable list view, when clicking on a cell which contains a field boolean